### PR TITLE
Fixing problem with parsing patient uuid from url

### PR DESCRIPTION
**Pull request by internettrans**
This fixes a bug where esm-api was incorrectly parsing the patient uuid from the url

### CODE REVIEW

File: src/shared-api-objects/current-patient.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    read: jest.fn()
+  }
+}));
+
+describe("current patient", () => {
+  beforeEach(() => {
+    fhir.read.mockReset();
+  });
+
+  it("fetches the correct patient from a patient chart URL", () => {
```
Comment (internettrans) [inline src/shared-api-objects/current-patient.test.ts:16]: This test fails without the code change in this PR

File: src/shared-api-objects/current-patient.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ window.addEventListener("single-spa:routing-event", () => {
 });
 
 function getPatientUuidFromUrl() {
-  const match = /\/patient\/(.+)\/?/.exec(location.pathname);
+  const match = /\/patient\/([a-zA-Z0-9\-]+)\/?/.exec(location.pathname);
```
Comment (internettrans) [inline src/shared-api-objects/current-patient.ts:20]: `+` is "greedy" (a regex term), so `.+` will actually capture the trailing `/chart` as part of the capture group. My solution is to whitelist which characters are allowed within patient uuids (notably not including the `/` character)

File: src/shared-api-objects/current-patient.test.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      .toPromise()
+      .then(() => {
+        expect(fhir.read).toHaveBeenCalledWith({
+          type: "Patient",
+          patient: "34"
+        });
+      });
+  });
+
+  it("can handle dashes and alphanumeric characters in the patient uuid", () => {
```
Comment (internettrans) [inline src/shared-api-objects/current-patient.test.ts:58]: This test fails without the code change in this PR

**Review by jonathandick — APPROVED** [review]
LGTM

**Pull request merged**